### DISCUSSION TITLE
Projects

### DISCUSSION THREAD

**Discussion by matari** [jbexta/AgentPilot#17] [Feature Requests]
It would be useful to be able to organize chat history and output files into projects. My use car is as follows, I have a project assistant agent that just has a conversation with me about my project idea. It is instructed to engage in a natural conversation with the user in order to gather all the information required to complete the project. Then the chat history is used by action agents to complete tasks required for completing the goal. The action agents could review the chat history and submit questions to the  interview. I wrote a very basic version of this and it works well. We love to talk about our projects..

**Comment by jbexta** [comment]
Hi @matari thanks for the suggestion! I have some features on the roadmap which might be useful for you.
The next breaking version (0.2.0) will have the ability to organise chats into renamable folders, which could be used to organise projects.

Another feature I'm thinking about adding is chat "tags", which could be automatically or manually added to each new chat (or maybe folders too?)

The context list page will have an option to filter by name, agent, folder and tags.

I'd be interested in learning more about your use case as it might influence the way I implement actions/tools 🙏

**Reply by matari** [reply]
Definitely useful feature adds. One of my use cases is content creation. The project type could be a book, video and podcast series. You would have a conversation about your idea with a single agent. This agent would have a system message instructing it to discuss the users project ina natural conversational tone and to ask questions in order to collect all the information required to build complete the project. Worker agents would read the conversation history and complete tasks like create an outline, distill the ideas I to notes, organize the  main points into the outline, write image descriptions, generate images, etc.. all of the output files would be saved to the project output folder and would also be embedded in a vector store to act as context. I'm thinking the actions could be triggered or run at fixed intervals.  A file viewer wysiwyg or code editor would allow the user to review the output of the worker agents and edit them or shift the conversation to that file. External files added to the project folder would be added as knowledge. 

So working on your project would be having a conversation and then editing.

**Reply by jbexta** [reply]
Ok interesting ideas, the initial 0.2.0 is a month or 2 away, I'm a solo dev working on it part time but this will be possible in 0.2.0 but there's so many possible configurations for complex workflows like that, I'm trying to allow as many configurations as possible. After it's released I'll put out a guide on making something like this in agent pilot.
I can see a path forward now, but there's a lot of work to do.
Interesting idea about shifting the conversation to a file, I'll have a think about how to implement that
